### PR TITLE
fix: fix MySQL stale connection error

### LINKED ISSUE #17324: [Bug]: "Secondary stale connection error" When Using MySQL Databases
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Some users that are using MySQL or MariaDB as datasource get often the error "Secondary stale connection error". Restarting the machine/containers fixes the issue, but is inconvenient for the users.

### Steps To Reproduce

https://app.frontapp.com/open/msg_1ea647ik
https://app.frontapp.com/open/msg_1e9wvc5o
https://app.frontapp.com/open/msg_1e3k7yh8

# Workaround
The following two things may help:
(1) Reduce the number of rows being fetched via [pagination](https://docs.appsmith.com/reference/widgets/table#server-side-pagination). It has been reported / noticed that fetching large number of rows can cause this error. 
(2) Restart the API server. 

### Public Sample App

_No response_

### Version

Self-Hosted v1.8.4

Front logo [Front conversations](https://app.frontapp.com/open/top_56b4c)

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- This PR is primarily meant to add changes to address the `StaleConnection` exception caused by MySQL plugin, however there are some minor refactor changes as well. 
- The following is the pattern in which the error shows itself: The queries work fine initially, upon checking the queries few hours later the queries fail with the `StaleConnection` error. If the application is left as it is then few minutes / hours later the queries start working fine again. This cycle keeps repeating. 
  - Based on the pattern with which the stale connection error shows itself (as reported in the linked issue), it seems reasonable to assume that the following could be happening: (1) at least one of the MySQL connections goes bad and holds up the execution for rest of the connections (2) at least one of the MySQL connection goes bad and holds up the common thread scheduler used by the MySQL plugin. In both of these cases, the issue is initiated by a bad connection. 
  - Based on the above assessment this PR changes some connection pool parameters for the MySQL plugin to make sure that the connections are refreshed frequently. **_A deploy preview with the changes was also successfully tested by the user who had reported the issue_**. However, it is important to note that even if the deploy preview was successful it may not mean success in production environment because the production instance has a much higher cumulative load. 
  - Apart from the change in connection parameters, the `StaleConnection` errors are updated to carry the upstream error to the client as well. So far one of the challenges with StaleConnection errors has been that it did not share the upstream error making it difficult to debug the root cause. 
  - Update MySQL driver version.
- Other refactor changes not related to the main issue: 
  - Explicit empty constructor definition is replaced with Lombok annotation for all error messages class. 
  - A base class is created for plugin error messages class to store all common error messages. 
  - Fix Indentation.

Screenshot 2023-06-22 at 3 53 14 PM

#### PR fixes following issue(s)
Fixes #17324 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

## Testing
- [x] Manual
- [x] JUnit TCs for some parts of code flow 
- Adding automated test for StaleConnection error does not seem possible at the moment because we have not been able to reproduce the error in dev / test environment.

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by nidhi-nair — APPROVED** [review]

File: app/server/appsmith-interfaces/pom.xml [hunk 1]
```diff
@@ -240,6 +240,36 @@
...
+            <artifactId>HikariCP</artifactId>
+            <version>5.0.1</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-api</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/pom.xml:254]: Hey sorry why is the interfaces module responsible for these dependencies? Can we please figure this out? We should never pull these out of the plugins module

File: app/server/appsmith-plugins/mssqlPlugin/pom.xml [hunk 2]
```diff
@@ -21,13 +21,15 @@
...
+            <scope>provided</scope>
         </dependency>
 
         <dependency>
             <groupId>com.zaxxer</groupId>
             <artifactId>HikariCP</artifactId>
+            <scope>provided</scope>
             <version>5.0.1</version>
-            <scope>compile</scope>
+<!--            <scope>compile</scope>-->
```
Comment (nidhi-nair) [inline app/server/appsmith-plugins/mssqlPlugin/pom.xml:32]: Can we remove this unused scope?

Comment (sumitsum) [reply app/server/appsmith-plugins/mssqlPlugin/pom.xml]: Done.

File: app/server/appsmith-interfaces/pom.xml (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-interfaces/pom.xml:254]: Continuing this thread on [Slack](https://theappsmith.slack.com/archives/C040LHZN03V/p1687759287264399?thread_ts=1687276128.909119&cid=C040LHZN03V).

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**